WeakAuras/RegionTypes/DynamicGroup.lua

```diff
@@ -409,6 +413,79 @@ local growers = {
       end
     end
   end,
+  GRID = function(data)
+    local gridType = data.gridType
+    local gridWidth = data.gridWidth
+    local rowSpace = data.rowSpace
+    local columnSpace = data.columnSpace
+    local rowFirst = (gridType:find("^[RLH]")) ~= nil
+    local limit = data.useLimit and data.limit or math.huge
+    local rowMul, colMul
+    if gridType:find("D") then
+      rowMul = -1
+    else
+      rowMul = 1
+    end
+    if gridType:find("L") then
+      colMul = -1
+    else
+      colMul = 1
+    end
+    return function(newPositions, activeRegions)
+      local numVisible = min(limit, #activeRegions)
+      local subLines = {}
+      local subLineIndex, startIndex = 0, 1 - gridWidth
+      while subLineIndex < numVisible / gridWidth do
```

Comment:
If I understood the effect of the algorithm correctly, there's a smaller and easier to understand algorithm. This is not 100% polished code, and I haven't even tested it. But the ideas should be clear from it.

First, let's define these somewhere outside of this block:

```
function GetWidth(child)
  return child.data.width or child.region.width
end

function GetHeight(child) 
  child.data.height or child.region.heigh
end

function SetX(newpositions, i, x)
  newpositions[i] =  newpositions[i] or {}
  newpositions[i].x = x
end

function SetY(newpositions, i, x)
  newpositions[i] =  newpositions[i] or {}
  newpositions[i].x = x
end
```
And then:
```
local GetFirst, GetSecond, SetFirst,SetSecond, firstSpacing, secondSpacing, firstMul, secondMul

if rowFirst then
  GetFirst, SetFirst, firstSpacing, firstMul = GetWidth, SetX, rowSpace, rowMul
  GetSecond, SetSecond, secondSpacing, secondMul = GetHeight, SetY, colSpace, colMul
else
   GetFirst, SetFirst, firstSpace, firstMul = GetHeight, SetY, colSpace, colMul
   GetSecond, SetSecond, secondSpace, secondMul = GetWidth, SetX, rowSpace, rowMul
end

local currentFirst = 0
local currentSecond = 0
local maxSecond = 0
for i = 1, numVisible do
  SetFirst(newPositions, i, currentFirst)
  SetSecond(newPositions, i, currentSecond)
  currentFirst = currentFirst + (firstSpace + GetFirst(activeRegions[i])) * firstMul
  maxSecond = max(masSecond, GetSecond(activeRegions[i])
  if i % gridWidth == 0 then
    currentSecond = currentSecond + (secondSpace + maxSecond) * secondMul
    currentFirst = 0
    maxSecond = 0
  end
end
```

Reply:
From a glance that looks very similar to my first implementation, which did not handle varied sizes well. The sublines construction is designed to ensure that children never overlap (outside of conditions changing position or size).

Reply:
If I understood your code correctly. this behaves exactly the same as your algorithm. Nothing ever overlaps with this.

Reply:
Well except it seems I swapped rowSpace and colSpace.

Reply:
That's possible, I haven't given your proposal any more than a glance.

And it's surprisingly easy to swap the spaces :)

Reply:
Also ```lua
function SetY(newpositions, i, x)
  newpositions[i] =  newpositions[i] or {}
  newpositions[i].x = x
end
```
is wrong 😄

Reply:
Ok, I examined it for more than 1 seconds and it looks good.